Userland/Applications/Browser/BrowserWindow.cpp

```diff
@@ -262,31 +263,40 @@ void BrowserWindow::build_menus()
     m_search_engine_actions.add_action(*m_disable_search_engine_action);
     m_disable_search_engine_action->set_checked(true);
 
-    auto search_engines_file = Core::File::construct(Browser::search_engines_file_path());
-    if (search_engines_file->open(Core::OpenMode::ReadOnly)) {
-        if (auto maybe_json = JsonValue::from_string(search_engines_file->read_all()); !maybe_json.is_error() && maybe_json.value().is_array()) {
-            auto json = maybe_json.release_value().as_array();
-            for (auto& json_item : json.values()) {
-                if (!json_item.is_object())
-                    continue;
-                auto search_engine = json_item.as_object();
-                auto name = search_engine.get("title").to_string();
-                auto url_format = search_engine.get("url_format").to_string();
-
-                auto action = GUI::Action::create_checkable(
-                    name, [&, url_format](auto&) {
-                        g_search_engine = url_format;
-                        Config::write_string("Browser", "Preferences", "SearchEngine", g_search_engine);
-                    },
-                    this);
-                search_engine_menu.add_action(action);
-                m_search_engine_actions.add_action(action);
-
-                if (g_search_engine == url_format) {
-                    action->set_checked(true);
-                    search_engine_set = true;
+    auto maybe_search_engines_file = Core::Stream::File::open(Browser::search_engines_file_path(), Core::Stream::OpenMode::Read);
+    if (!maybe_search_engines_file.is_error()) {
+        auto& search_engines_file = maybe_search_engines_file.value();
+        auto file_size = search_engines_file->size().release_value_but_fixme_should_propagate_errors();
+        auto maybe_buffer = ByteBuffer::create_uninitialized(file_size);
+        if (maybe_buffer.has_value()) {
```

Comment:
In these cases I just don't bother wrapping in an if and just `.release_value()` instead, with the hope that eventually we'll get around to having it be an ErrorOr API and fix the billion usages. :^)

Reply:
I did that initially but didn't like that release_value() hides that there should be an error check - someone who comes along later might not notice, whereas this makes it clear. I agree on ByteBuffer though, maybe I should get on and make that return errors since it was bothering me.